File: rxapp/src/main/java/justin/rxapp/ActivityResult.java
package justin.rxapp;

import android.app.Activity;
import android.content.Intent;

/**
 * @author justin on 2017/04/19 13:29
 * @version V1.0
 */
public class ActivityResult {
    private final int code;
    private final Intent data;

    public ActivityResult(int code, Intent data) {
        this.code = code;
        this.data = data;
    }

    public int getCode() {
        return code;
    }

    public Intent getData() {
        return data;
    }

    public boolean isOk() {
        return code == Activity.RESULT_OK;
    }

    public boolean isCancel() {
        return code == Activity.RESULT_CANCELED;
    }

    public boolean isFirstUser() {
        return code == Activity.RESULT_FIRST_USER;
    }

    @Override
    public String toString() {
        return "ActivityResult{" +
                "code=" + code +
                ", data=" + data +
                '}';
    }
}

File: rxapp-kotlin/src/main/java/justin/rxapp/ActivityResult.kt
package justin.rxapp

import android.app.Activity
import android.content.Intent

/**
 * @author justin on 2017/04/19 13:29
 * *
 * @version V1.0
 */
data class ActivityResult(val code: Int, val data: Intent) {

    val isOk: Boolean
        get() = code == Activity.RESULT_OK

    val isCancel: Boolean
        get() = code == Activity.RESULT_CANCELED

    val isFirstUser: Boolean
        get() = code == Activity.RESULT_FIRST_USER
}
